docusaurus/src/components/RequestERD/EmailModal.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import styles from "./EmailModal.module.css";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import cn from "classnames";
+import { useDebouncedCallback } from "use-debounce";
+import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
+
+const CloseButton = ({ onClose }) => {
+  return (
+    <button onClick={onClose} className={styles.modalContentClose}>
+      <FontAwesomeIcon icon={faXmark} />
+    </button>
+  );
+};
+
+const ModalBody = ({ status, onSubmit }) => {
+  const [email, setEmail] = useState("");
+  const [emailError, setEmailError] = useState("");
+
+  const debouncedValidation = useDebouncedCallback((value) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (emailRegex.test(value)) {
+      setEmailError("");
+    } else {
+      setEmailError("Please enter a valid email address");
+    }
+  }, 800);
```

Comment:
I don't think a debounce provides the best UX here, as it makes the validation feel laggy.

Instead, I think a better UX would be:
- Perform the validation on every change to the input
- Only show the error if the user blurs the input while there is a value in it. This way we give them a chance to write a full without showing any errors, even if they pause between some of the characters, and only show the error if they then leave the input (e.g. by clicking the Submit button)

Reply:
Great suggestion. Definitely better UX